diff --git a/beacon.c b/beacon.c
index e09c05c..0ccf422 100644
--- a/beacon.c
+++ b/beacon.c
@@ -293,6 +293,9 @@ void beacon_init (struct audio_s *pmodem, struct misc_config_s *pconfig, struct
 		  }
 		  break;
 
+                case BEACON_STATS:
+                  break;
+
 	        case BEACON_IGNORE:
 		  break;
 	      }
@@ -1017,6 +1020,29 @@ static void beacon_send (int j, dwgps_info_t *gpsinfo)
 	          }
 		  break;
 
+		case BEACON_STATS:
+
+                  {
+                    int last_minutes = 60;
+
+                    FILE *statFile = fopen("/tmp/dw-stats.txt", "w");
+                    if (statFile == NULL)
+                        {
+                            dw_printf("Error opening stats file!\n");
+                        }
+
+                    fprintf (statFile, "MSG_CNT=%d\r\nPKT_CNT=%d\r\nDIR_CNT=%d\r\nLOC_CNT=%d\r\nRF_CNT=%d\r\nUPL_CNT=%d\r\nDNL_CNT=%d\r\n",
+                                                igate_get_msg_cnt(),
+                                                igate_get_pkt_cnt(),
+                                                mheard_count(0,last_minutes),
+                                                mheard_count(g_igate_config_p->max_digi_hops,last_minutes),
+                                                mheard_count(8,last_minutes),
+                                                igate_get_upl_cnt(),
+                                                igate_get_dnl_cnt());
+                    fclose(statFile);
+
+                  }
+		  break;
 		case BEACON_IGNORE:		
 	        default:
 		  break;
diff --git a/config.c b/config.c
index 9d6c083..9d3f709 100644
--- a/config.c
+++ b/config.c
@@ -4618,6 +4618,7 @@ void config_init (char *fname, struct audio_s *p_audio_config,
 		   strcasecmp(t, "OBEACON") == 0 ||
 		   strcasecmp(t, "TBEACON") == 0 ||
 		   strcasecmp(t, "CBEACON") == 0 ||
+		   strcasecmp(t, "SBEACON") == 0 ||
 		   strcasecmp(t, "IBEACON") == 0) {
 
 	    if (p_misc_config->num_beacons < MAX_BEACONS) {
@@ -4635,6 +4636,9 @@ void config_init (char *fname, struct audio_s *p_audio_config,
 	      else if (strcasecmp(t, "IBEACON") == 0) {
 	        p_misc_config->beacon[p_misc_config->num_beacons].btype = BEACON_IGATE;
 	      }
+	      else if (strcasecmp(t, "SBEACON") == 0) {
+	        p_misc_config->beacon[p_misc_config->num_beacons].btype = BEACON_STATS;
+	      }
 	      else {
 	        p_misc_config->beacon[p_misc_config->num_beacons].btype = BEACON_CUSTOM;
 	      }
diff --git a/config.h b/config.h
index baf6cb4..b235604 100644
--- a/config.h
+++ b/config.h
@@ -24,7 +24,7 @@
  * This wasn't thought out.  It just happened.
  */
 
-enum beacon_type_e { BEACON_IGNORE, BEACON_POSITION, BEACON_OBJECT, BEACON_TRACKER, BEACON_CUSTOM, BEACON_IGATE };
+enum beacon_type_e { BEACON_IGNORE, BEACON_POSITION, BEACON_OBJECT, BEACON_TRACKER, BEACON_CUSTOM, BEACON_IGATE, BEACON_STATS };
 
 enum sendto_type_e { SENDTO_XMIT, SENDTO_IGATE, SENDTO_RECV };
 
